### PR TITLE
Session support for RBAC

### PR DESCRIPTION
**Pull request by julisch94**
- Implemented a custom role manager for RBAC with sessions.
- Also added documentation to the code and included two simple examples: one model example and one policy example.

Not able to write tests because my compiler doesn't find `t.Helper()`.

Targetting https://github.com/casbin/casbin/issues/43

### CODE REVIEW

**Comment by hsluoyz** [comment]
Thanks for the quick PR! An issue is the test is missing. ``t.Helper()`` is supported in Go 1.9. Could you upgrade to Go 1.9.1?

**Comment by julisch94** [comment]
I'll do that and add some tests by the end of next week.

**Comment by julisch94** [comment]
Could you please make the function GetUsers(name string) in RoleManager also accept multiple arguments? Because I'd like to use it as "Who is using role r at time t?", which will only be possible if I can pass one more argument to GetUsers.

That would be great!

**Comment by hsluoyz** [comment]
Added here: https://github.com/casbin/casbin/commit/a57b2b7e0f2c28d42ddcef2d98f9b37c83394d47

**Comment by julisch94** [comment]
I have lately added some tests with coverage above 80%.

File: rbac/session_role_manager.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package rbac
 
 import (
 	"github.com/casbin/casbin/util"
+	"sort"
```
Comment (hsluoyz) [inline rbac/session_role_manager.go]: Please put the built-in imports before github imports.

**Comment by hsluoyz** [comment]
@julisch94  the test failed:

```
	session_role_manager_test.go:48: alpha < bravo at time 1509106078517347100: true, supposed to be false
	session_role_manager_test.go:49: alpha < charlie at time 1509106078517347100: true, supposed to be false
	session_role_manager_test.go:50: bravo < delta at time 1509106078517347100: true, supposed to be false
	session_role_manager_test.go:51: bravo < echo at time 1509106078517347100: true, supposed to be false
	session_role_manager_test.go:52: charlie < echo at time 1509106078517347100: true, supposed to be false
	session_role_manager_test.go:53: charlie < foxtrott at time 1509106078517347100: true, supposed to be false
```

**Comment by julisch94** [comment]
I see you made a few improvements and an import fix. Do these happen to fix the test fails you mentioned? Because having pulled your latest changes, I can't reproduce the failing tests.

**Comment by hsluoyz** [comment]
Yeah, I fixed them, no issues now. 👍

**Pull request merged**